File: ltsm/pythonProject/predictCrash/dtos.py
""" Options:
Date: 2024-09-16 17:15:28
Version: 8.40
Tip: To override a DTO option, remove "#" prefix before updating
BaseUrl: https://crash.digitalstar.co

#GlobalNamespace: 
#AddServiceStackTypes: True
#AddResponseStatus: False
#AddImplicitVersion: 
#AddDescriptionAsComments: True
#IncludeTypes: 
#ExcludeTypes: 
#DefaultImports: datetime,decimal,marshmallow.fields:*,servicestack:*,typing:*,dataclasses:dataclass/field,dataclasses_json:dataclass_json/LetterCase/Undefined/config,enum:Enum/IntEnum
#DataClass: 
#DataClassJson: 
"""

import datetime
import decimal
from marshmallow.fields import *
from servicestack import *
from typing import *
from dataclasses import dataclass, field
from dataclasses_json import dataclass_json, LetterCase, Undefined, config
from enum import Enum, IntEnum


@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class ActiveGameRoomPrediction:
    id: int = 0
    game_number: int = 0
    room_id: int = 0
    active_game_room_id: Optional[int] = None
    round_id: int = 0
    prediction: Decimal = decimal.Decimal(0)
    prediction2: Decimal = decimal.Decimal(0)
    prediction3: Decimal = decimal.Decimal(0)
    prediction4: Decimal = decimal.Decimal(0)
    prediction_arima: Decimal = decimal.Decimal(0)


@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class AppUser:
    id: Optional[str] = None
    first_name: Optional[str] = None
    user_name: Optional[str] = None
    last_name: Optional[str] = None
    display_name: Optional[str] = None


@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class ApplicationUserPaymentLog:
    id: int = 0
    payment_covers_from: datetime.datetime = datetime.datetime(1, 1, 1)
    payment_covers_until: datetime.datetime = datetime.datetime(1, 1, 1)
    cost_usd: Decimal = decimal.Decimal(0)
    cost_sol: Decimal = decimal.Decimal(0)
    # @References(typeof(AppUser))
    users_id: Optional[str] = None

    application_app_user: Optional[AppUser] = None
    transaction_hash: Optional[str] = None
    notes: Optional[str] = None


@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class ActiveGameRoom:
    id: int = 0
    game_number: int = 0
    room_id: int = 0
    round_id: int = 0
    # @StringLength(255)
    game_status: Optional[str] = None

    # @StringLength(255)
    game_phase: Optional[str] = None

    game_result: Decimal = decimal.Decimal(0)
    no_more_bets_at: int = 0
    time_recorded: int = 0


@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class ActiveGameResult:
    active_game_room_prediction_id: int = 0
    game_number: int = 0
    room_id: int = 0
    active_game_room_round_id: int = 0
    game_result: Decimal = decimal.Decimal(0)
    prediction: Decimal = decimal.Decimal(0)
    prediction2: Decimal = decimal.Decimal(0)
    prediction3: Decimal = decimal.Decimal(0)
    prediction4: Decimal = decimal.Decimal(0)
    prediction_arima: Decimal = decimal.Decimal(0)
    no_more_bets_at: int = 0


@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class NotifyLiveViewDataUpdatedResponse:
    pass


# @Route("/notify")
@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class NotifyLiveViewDataUpdatedRequest(IReturn[NotifyLiveViewDataUpdatedResponse]):
    id: int = 0
    tea_cup: bool = False


# @Route("/data/queryactivegameroomprediction")
@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class ApiQueryActiveGameRoomPrediction(QueryDb[ActiveGameRoomPrediction], IReturn[QueryResponse[ActiveGameRoomPrediction]]):
    id: Optional[int] = None


# @Route("/query/predicts")
@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class QueryActiveGameRoomPrediction(QueryDb[ActiveGameRoomPrediction], IReturn[QueryResponse[ActiveGameRoomPrediction]]):
    id: Optional[int] = None
    round_id: Optional[int] = None
    round_ids: Optional[List[int]] = None


@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class QueryApplicationUserPaymentLog(QueryDb[ApplicationUserPaymentLog], IReturn[QueryResponse[ApplicationUserPaymentLog]]):
    id: Optional[int] = None
    payment_covers_from: Optional[datetime.datetime] = None
    payment_covers_until: Optional[datetime.datetime] = None
    application_user_id: Optional[str] = None


@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class ApiQueryActiveGameRoom(QueryDb[ActiveGameRoom], IReturn[QueryResponse[ActiveGameRoom]]):
    id: Optional[int] = None
    round_id: Optional[int] = None
    room_id: Optional[int] = None
    time_recorded: Optional[int] = None


# @Route("/query/activeroom", "GET")
@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class QueryActiveGameRoom(QueryDb[ActiveGameRoom], IReturn[QueryResponse[ActiveGameRoom]]):
    id: Optional[int] = None
    round_id: Optional[int] = None
    room_id: Optional[int] = None
    time_recorded: Optional[int] = None


@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class ApiQueryActiveRoomPredictionResults(QueryDb2[ActiveGameRoomPrediction, ActiveGameResult], IReturn[QueryResponse[ActiveGameResult]]):
    pass


# @Route("/query/activeroomprediction", "GET")
@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class QueryActiveRoomPredictionResults(QueryDb2[ActiveGameRoomPrediction, ActiveGameResult], IReturn[QueryResponse[ActiveGameResult]]):
    pass


@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class CreateActiveGameRoomPrediction(IReturn[IdResponse], ICreateDb[ActiveGameRoomPrediction]):
    game_number: int = 0
    room_id: int = 0
    round_id: int = 0
    prediction: Decimal = decimal.Decimal(0)
    prediction2: Decimal = decimal.Decimal(0)
    prediction3: Decimal = decimal.Decimal(0)
    prediction4: Decimal = decimal.Decimal(0)
    prediction_arima: Decimal = decimal.Decimal(0)
    active_game_room_id: Optional[int] = None


@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class CreateApplicationUserPaymentLog(IReturn[IdResponse], ICreateDb[ApplicationUserPaymentLog]):
    pass


# @Route("/createdata")
@dataclass_json(letter_case=LetterCase.CAMEL, undefined=Undefined.EXCLUDE)
@dataclass
class CreateActiveGameRoom(IReturn[IdResponse], ICreateDb[ActiveGameRoom]):
    game_number: int = 0
    room_id: int = 0
    round_id: int = 0
    game_status: Optional[str] = None
    game_phase: Optional[str] = None
    game_result: Decimal = decimal.Decimal(0)
    no_more_bets_at: int = 0
    time_recorded: int = 0

File: CrashServer/wwwroot/mjs/dtos.mjs
/* Options:
Date: 2024-09-16 13:44:23
Version: 8.40
Tip: To override a DTO option, remove "//" prefix before updating
BaseUrl: https://localhost:5001

//AddServiceStackTypes: True
//AddDocAnnotations: True
//AddDescriptionAsComments: True
//IncludeTypes: 
//ExcludeTypes: 
//DefaultImports: 
*/

"use strict";
export class QueryBase {
    /** @param {{skip?:number,take?:number,orderBy?:string,orderByDesc?:string,include?:string,fields?:string,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {?number} */
    skip;
    /** @type {?number} */
    take;
    /** @type {string} */
    orderBy;
    /** @type {string} */
    orderByDesc;
    /** @type {string} */
    include;
    /** @type {string} */
    fields;
    /** @type {{ [index: string]: string; }} */
    meta;
}
/** @typedef T {any} */
export class QueryDb_1 extends QueryBase {
    /** @param {{skip?:number,take?:number,orderBy?:string,orderByDesc?:string,include?:string,fields?:string,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { super(init); Object.assign(this, init) }
}
export class ActiveGameRoomPrediction {
    /** @param {{id?:number,gameNumber?:number,roomId?:number,activeGameRoomId?:number,roundId?:number,prediction?:number,prediction2?:number,prediction3?:number,prediction4?:number,predictionArima?:number}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {number} */
    id;
    /** @type {number} */
    gameNumber;
    /** @type {number} */
    roomId;
    /** @type {?number} */
    activeGameRoomId;
    /** @type {number} */
    roundId;
    /** @type {number} */
    prediction;
    /** @type {number} */
    prediction2;
    /** @type {number} */
    prediction3;
    /** @type {number} */
    prediction4;
    /** @type {number} */
    predictionArima;
}
export class AppUser {
    /** @param {{id?:string,firstName?:string,userName?:string,lastName?:string,displayName?:string}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {string} */
    id;
    /** @type {?string} */
    firstName;
    /** @type {?string} */
    userName;
    /** @type {?string} */
    lastName;
    /** @type {?string} */
    displayName;
}
export class ApplicationUserPaymentLog {
    /** @param {{id?:number,paymentCoversFrom?:string,paymentCoversUntil?:string,costUsd?:number,costSol?:number,usersId?:string,applicationAppUser?:AppUser,transactionHash?:string,notes?:string}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {number} */
    id;
    /** @type {string} */
    paymentCoversFrom;
    /** @type {string} */
    paymentCoversUntil;
    /** @type {number} */
    costUsd;
    /** @type {number} */
    costSol;
    /** @type {?string} */
    usersId;
    /** @type {AppUser} */
    applicationAppUser;
    /** @type {string} */
    transactionHash;
    /** @type {string} */
    notes;
}
export class ActiveGameRoom {
    /** @param {{id?:number,gameNumber?:number,roomId?:number,roundId?:number,gameStatus?:string,gamePhase?:string,gameResult?:number,noMoreBetsAt?:number,timeRecorded?:number}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {number} */
    id;
    /** @type {number} */
    gameNumber;
    /** @type {number} */
    roomId;
    /** @type {number} */
    roundId;
    /** @type {string} */
    gameStatus;
    /** @type {string} */
    gamePhase;
    /** @type {number} */
    gameResult;
    /** @type {number} */
    noMoreBetsAt;
    /** @type {?number} */
    timeRecorded;
}
/** @typedef From {any} */
/** @typedef  Into {any} */
export class QueryDb_2 extends QueryBase {
    /** @param {{skip?:number,take?:number,orderBy?:string,orderByDesc?:string,include?:string,fields?:string,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { super(init); Object.assign(this, init) }
}
export class ActiveGameResult {
    /** @param {{id?:number,gameNumber?:number,roomId?:number,activeGameRoomRoundId?:number,gameResult?:number,prediction?:number,prediction2?:number,prediction3?:number,prediction4?:number,predictionArima?:number,noMoreBetsAt?:number}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {number} */
    id;
    /** @type {number} */
    gameNumber;
    /** @type {number} */
    roomId;
    /** @type {number} */
    activeGameRoomRoundId;
    /** @type {number} */
    gameResult;
    /** @type {number} */
    prediction;
    /** @type {number} */
    prediction2;
    /** @type {number} */
    prediction3;
    /** @type {number} */
    prediction4;
    /** @type {number} */
    predictionArima;
    /** @type {number} */
    noMoreBetsAt;
}
export class ResponseError {
    /** @param {{errorCode?:string,fieldName?:string,message?:string,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {string} */
    errorCode;
    /** @type {string} */
    fieldName;
    /** @type {string} */
    message;
    /** @type {{ [index: string]: string; }} */
    meta;
}
export class ResponseStatus {
    /** @param {{errorCode?:string,message?:string,stackTrace?:string,errors?:ResponseError[],meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {string} */
    errorCode;
    /** @type {string} */
    message;
    /** @type {string} */
    stackTrace;
    /** @type {ResponseError[]} */
    errors;
    /** @type {{ [index: string]: string; }} */
    meta;
}
export class NotifyLiveViewDataUpdatedResponse {
    constructor(init) { Object.assign(this, init) }
}
export class AuthenticateResponse {
    /** @param {{userId?:string,sessionId?:string,userName?:string,displayName?:string,referrerUrl?:string,bearerToken?:string,refreshToken?:string,refreshTokenExpiry?:string,profileUrl?:string,roles?:string[],permissions?:string[],responseStatus?:ResponseStatus,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {string} */
    userId;
    /** @type {string} */
    sessionId;
    /** @type {string} */
    userName;
    /** @type {string} */
    displayName;
    /** @type {string} */
    referrerUrl;
    /** @type {string} */
    bearerToken;
    /** @type {string} */
    refreshToken;
    /** @type {?string} */
    refreshTokenExpiry;
    /** @type {string} */
    profileUrl;
    /** @type {string[]} */
    roles;
    /** @type {string[]} */
    permissions;
    /** @type {ResponseStatus} */
    responseStatus;
    /** @type {{ [index: string]: string; }} */
    meta;
}
/** @typedef T {any} */
export class QueryResponse {
    /** @param {{offset?:number,total?:number,results?:T[],meta?:{ [index: string]: string; },responseStatus?:ResponseStatus}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {number} */
    offset;
    /** @type {number} */
    total;
    /** @type {T[]} */
    results;
    /** @type {{ [index: string]: string; }} */
    meta;
    /** @type {ResponseStatus} */
    responseStatus;
}
export class IdResponse {
    /** @param {{id?:string,responseStatus?:ResponseStatus}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {string} */
    id;
    /** @type {ResponseStatus} */
    responseStatus;
}
export class NotifyLiveViewDataUpdatedRequest {
    /** @param {{id?:number,teaCup?:boolean}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {number} */
    id;
    /** @type {boolean} */
    teaCup;
    getTypeName() { return 'NotifyLiveViewDataUpdatedRequest' }
    getMethod() { return 'POST' }
    createResponse() { return new NotifyLiveViewDataUpdatedResponse() }
}
export class Authenticate {
    /** @param {{provider?:string,userName?:string,password?:string,rememberMe?:boolean,accessToken?:string,accessTokenSecret?:string,returnUrl?:string,errorView?:string,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { Object.assign(this, init) }
    /**
     * @type {string}
     * @description AuthProvider, e.g. credentials */
    provider;
    /** @type {string} */
    userName;
    /** @type {string} */
    password;
    /** @type {?boolean} */
    rememberMe;
    /** @type {string} */
    accessToken;
    /** @type {string} */
    accessTokenSecret;
    /** @type {string} */
    returnUrl;
    /** @type {string} */
    errorView;
    /** @type {{ [index: string]: string; }} */
    meta;
    getTypeName() { return 'Authenticate' }
    getMethod() { return 'POST' }
    createResponse() { return new AuthenticateResponse() }
}
export class ApiQueryActiveGameRoomPrediction extends QueryDb_1 {
    /** @param {{id?:number,skip?:number,take?:number,orderBy?:string,orderByDesc?:string,include?:string,fields?:string,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { super(init); Object.assign(this, init) }
    /** @type {?number} */
    id;
    getTypeName() { return 'ApiQueryActiveGameRoomPrediction' }
    getMethod() { return 'GET' }
    createResponse() { return new QueryResponse() }
}
export class QueryActiveGameRoomPrediction extends QueryDb_1 {
    /** @param {{id?:number,roundId?:number,roundIds?:number[],skip?:number,take?:number,orderBy?:string,orderByDesc?:string,include?:string,fields?:string,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { super(init); Object.assign(this, init) }
    /** @type {?number} */
    id;
    /** @type {?number} */
    roundId;
    /** @type {number[]} */
    roundIds;
    getTypeName() { return 'QueryActiveGameRoomPrediction' }
    getMethod() { return 'GET' }
    createResponse() { return new QueryResponse() }
}
export class QueryApplicationUserPaymentLog extends QueryDb_1 {
    /** @param {{id?:number,paymentCoversFrom?:string,paymentCoversUntil?:string,applicationUserId?:string,skip?:number,take?:number,orderBy?:string,orderByDesc?:string,include?:string,fields?:string,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { super(init); Object.assign(this, init) }
    /** @type {?number} */
    id;
    /** @type {?string} */
    paymentCoversFrom;
    /** @type {?string} */
    paymentCoversUntil;
    /** @type {?string} */
    applicationUserId;
    getTypeName() { return 'QueryApplicationUserPaymentLog' }
    getMethod() { return 'GET' }
    createResponse() { return new QueryResponse() }
}
export class ApiQueryActiveGameRoom extends QueryDb_1 {
    /** @param {{id?:number,roundId?:number,roomId?:number,timeRecorded?:number,skip?:number,take?:number,orderBy?:string,orderByDesc?:string,include?:string,fields?:string,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { super(init); Object.assign(this, init) }
    /** @type {?number} */
    id;
    /** @type {?number} */
    roundId;
    /** @type {?number} */
    roomId;
    /** @type {?number} */
    timeRecorded;
    getTypeName() { return 'ApiQueryActiveGameRoom' }
    getMethod() { return 'GET' }
    createResponse() { return new QueryResponse() }
}
export class QueryActiveGameRoom extends QueryDb_1 {
    /** @param {{id?:number,roundId?:number,roomId?:number,timeRecorded?:number,skip?:number,take?:number,orderBy?:string,orderByDesc?:string,include?:string,fields?:string,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { super(init); Object.assign(this, init) }
    /** @type {?number} */
    id;
    /** @type {?number} */
    roundId;
    /** @type {?number} */
    roomId;
    /** @type {?number} */
    timeRecorded;
    getTypeName() { return 'QueryActiveGameRoom' }
    getMethod() { return 'GET' }
    createResponse() { return new QueryResponse() }
}
export class ApiQueryActiveRoomPredictionResults extends QueryDb_2 {
    /** @param {{skip?:number,take?:number,orderBy?:string,orderByDesc?:string,include?:string,fields?:string,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { super(init); Object.assign(this, init) }
    getTypeName() { return 'ApiQueryActiveRoomPredictionResults' }
    getMethod() { return 'GET' }
    createResponse() { return new QueryResponse() }
}
export class QueryActiveRoomPredictionResults extends QueryDb_2 {
    /** @param {{skip?:number,take?:number,orderBy?:string,orderByDesc?:string,include?:string,fields?:string,meta?:{ [index: string]: string; }}} [init] */
    constructor(init) { super(init); Object.assign(this, init) }
    getTypeName() { return 'QueryActiveRoomPredictionResults' }
    getMethod() { return 'GET' }
    createResponse() { return new QueryResponse() }
}
export class CreateActiveGameRoomPrediction {
    /** @param {{gameNumber?:number,roomId?:number,roundId?:number,prediction?:number,prediction2?:number,prediction3?:number,prediction4?:number,predictionArima?:number,activeGameRoomId?:number}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {number} */
    gameNumber;
    /** @type {number} */
    roomId;
    /** @type {number} */
    roundId;
    /** @type {number} */
    prediction;
    /** @type {number} */
    prediction2;
    /** @type {number} */
    prediction3;
    /** @type {number} */
    prediction4;
    /** @type {number} */
    predictionArima;
    /** @type {?number} */
    activeGameRoomId;
    getTypeName() { return 'CreateActiveGameRoomPrediction' }
    getMethod() { return 'POST' }
    createResponse() { return new IdResponse() }
}
export class CreateApplicationUserPaymentLog {
    constructor(init) { Object.assign(this, init) }
    getTypeName() { return 'CreateApplicationUserPaymentLog' }
    getMethod() { return 'POST' }
    createResponse() { return new IdResponse() }
}
export class CreateActiveGameRoom {
    /** @param {{gameNumber?:number,roomId?:number,roundId?:number,gameStatus?:string,gamePhase?:string,gameResult?:number,noMoreBetsAt?:number,timeRecorded?:number}} [init] */
    constructor(init) { Object.assign(this, init) }
    /** @type {number} */
    gameNumber;
    /** @type {number} */
    roomId;
    /** @type {number} */
    roundId;
    /** @type {string} */
    gameStatus;
    /** @type {string} */
    gamePhase;
    /** @type {number} */
    gameResult;
    /** @type {number} */
    noMoreBetsAt;
    /** @type {number} */
    timeRecorded;
    getTypeName() { return 'CreateActiveGameRoom' }
    getMethod() { return 'POST' }
    createResponse() { return new IdResponse() }
}

